test/extensions/filters/http/fault/fault_filter_test.cc

```diff
@@ -667,6 +688,7 @@ TEST_F(FaultFilterTest, ConsecutiveDelayFaults) {
 
   // Have the fault delay of request 2 kick in, which should be delayed with success.
   timer_->invokeCallback();
+  EXPECT_THAT(dynamic_metadata, ProtoEq(config_->settings()->metadata()));
```

Comment:
After your recent updates, here need to call `filterMetadata()` which returns `Envoy::ProtobufWkt::Struct`

You could just get `ProtobufWkt::Struct` from dynamic_metadata and  test equality.